twilight-gateway/src/shard.rs

```diff
@@ -558,17 +562,17 @@ impl<Q> Shard<Q> {
                 ReceiveMessageError::from_websocket(source)
             }))?;
 
-        let Some(pending) = &mut self.pending else {
-            return Poll::Ready(Ok(()));
-        };
```

Comment:
This saved an unwrap that was now added, why remove it?

Reply:
By borrowing `self.pending` it's no longer possible to use `map_error` (which borrows `self`), forcing the use of verbose if let error-handling statements.